simd/src/Kokkos_Simd_Common.hpp

```diff
@@ -0,0 +1,224 @@
+#ifndef KOKKOS_SIMD_COMMON_HPP
+#define KOKKOS_SIMD_COMMON_HPP
+
+#include <cmath>
+#include <cstring>
+
+#include <Kokkos_Macros.hpp>
+
+#include <desul/atomics/Common.hpp>
+
+namespace Kokkos {
+namespace Experimental {
+
+template <class To, class From>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr To bit_cast(
+    From const& src) {
+  To dst;
+  std::memcpy(&dst, &src, sizeof(To));
+  return dst;
+}
+
+template <class T, class Abi>
+class simd;
+
+template <class T, class Abi>
+class simd_mask;
+
+class element_aligned_tag {};
```

Comment:
Prefer
```suggestion
struct element_aligned_tag {};
```
for tag classes.